examples/recognition_pipeline/recognition_pipeline.py

```diff
@@ -0,0 +1,198 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+ 
+    https://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+An Image Recognition pipeline to label images from specific dataset by a specific provider.
+
+The pipeline uses Python's Apache beam library to parralize the different stages.
+The images are taken from a Firestore database and are labeled by a ML provider.
+The labeling content is updated in the database for each image.
+By the end of the process, the project's admin group get notified.
+"""
+
+from __future__ import absolute_import
+
+import argparse
+import logging
+import random
+from datetime import datetime
+
+from past.builtins import unicode
+
+from firebase_admin import firestore
+
+import apache_beam as beam
+from apache_beam.io import WriteToText
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+
+from providers import google_vision_api
+from pipeline_lib.image_filtering import is_eligible
+from pipeline_lib.redefine_labels import RedefineLabels
+
+RANGE_OF_BATCH = 10 # Defines the range of the each batch for querying the database.
+collection_name = 'Images'
+
+def initialize_DB(): 
+    """Initializes project's Firestore databse for writing and reading purposes.
+
+    """
+    if not firebase_admin._apps:
+        firebase_admin.initialize_app(credentials.ApplicationDefault(), {
+        'projectId': 'step-project-ellispis',
+        })
+    return firestore.client()
+
+def get_provider(provider_name):
+    """ Creates an object of type ImageRecognitionProvider by the specific provider input.
+
+    """ 
+    if provider_name == 'Google_Vision_API':
+        return google_vision_api.GoogleVisionAPI()
+    
+class GetDataset(beam.DoFn):
+    """Queries the project's database to get the image dataset to label.
+
```

Comment:
Rename to `GetBatchedDataset`? As it returns the data split into batches for parallelization

Reply:
renamed.